doc-content/shared-kie-docs/src/main/asciidoc/Workbench/RemoteAPI/security-management-rest-api-users-ref.adoc

```diff
@@ -0,0 +1,255 @@
+
+[GET] /users::
++
+--
+Returns all users in {CENTRAL}.
+
+.Example server response (JSON)
+[source,json]
+----
+[
+    "newUser",
+    "user1",
+    "user2",
+]
+----
+--
+
+[GET] /users/{userName}/groups::
++
+--
+Returns all groups assigned to a specified user.
+
+.Request parameters
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`userName`
+|Name of the user for whom you are retrieving assigned groups
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+[{
+		"group1"
+	},
+	{
+		"group2"
+	}
+]
+----
+--
+
+[GET] /users/{userName}/roles::
++
+--
+Returns all roles assigned to a specified user.
+
+.Request parameters
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`userName`
+|Name of the user for whom you are retrieving assigned roles
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+[
+  {
+    "name": "process-admin"
+  },
+  {
+    "name": "manager"
+  },
+  {
+    "name": "admin"
+  }
+]
+----
+--
+
+[POST] /users::
++
+--
+Creates a specified user with specified roles and groups.
+
+.Example request body (JSON)
+[source,json]
+----
+{
+  "name": "newUser",
+  "roles": [
+    "roles.."
+  ],
+  "groups": [
+    "groups.."
+  ]
+}
+----
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "status": "OK",
+  "message": "User newUser is created successfully."
+}
+----
+--
+
+[Post] /users/{userName}/changePassword::
++
+--
+Changes the password of a specified user.
+
+.Request parameters
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`userName`
+|Name of the user for whom you are changing the password
+|String
+|Required
+|===
+
+.Example request command
+[source,subs="attributes+"]
+----
+curl -u 'baAdmin:password@1' -H "accept: application/json" -H "content-type: application/json" -X POST "http://localhost:8080/{URL_COMPONENT_CENTRAL}/rest/users/newUser/changePassword" -d newpassword
+----
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "status": "OK",
+  "message": "Password for newUser has been updated successfully."
+}
+----
+--
+
+[DELETE] /users/{userName}::
++
+--
+Deletes a specified user from {CENTRAL}.
+
+.Request parameters
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`userName`
+|Name of the user to be deleted
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "status": "OK",
+  "message": "User newUser is deleted successfully."
+}
+----
+--
+
+[POST] /users/{userName}/groups::
++
+--
+Updates the groups assigned to a specified user.
```

Comment:
Maybe it would be good to write if it just adds the groups to the existing ones or it will override all of them. Not sure what is the case here.

Reply:
Nice observation, I think it will be good to add that. 
@hmanwani-rh All the update API's overrides the current value with new ones. Only the Update permission API's for groups and roles add permissions to existing ones.

Reply:
@MarianMacik and @RishiRajAnand
How about changing to the following:
 "Overrides the existing groups with new ones assigned to a specified user."

Reply:
Fine with me.

Reply:
Thanks, changing the description for roles too.

Reply:
@MarianMacik and @RishiRajAnand
I have updated the sections as per the suggested changes. Please review and let me know if you need any other changes to be done.